src/meta/TableSpec.h

```diff
@@ -207,14 +207,22 @@ constexpr auto DAY_HOURS = 24;
 constexpr auto HOUR_SECONDS = HOUR_MINUTES * MINUTE_SECONDS;
 constexpr auto DAY_SECONDS = HOUR_SECONDS * DAY_HOURS;
 
-const nebula::common::unordered_map<nebula::meta::PatternMacro, std::string> patternStr{
+const nebula::common::unordered_map<nebula::meta::PatternMacro, std::string> patternYMLStr{
   { nebula::meta::PatternMacro::DATE, "/{date/}" },
   { nebula::meta::PatternMacro::HOUR, "/{hour/}" },
   { nebula::meta::PatternMacro::MINUTE, "/{minute/}" },
   { nebula::meta::PatternMacro::SECOND, "/{second/}" },
   { nebula::meta::PatternMacro::TIMESTAMP, "/{timestamp/}" }
 };
 
+const nebula::common::unordered_map<nebula::meta::PatternMacro, std::string> patternJSONStr{
+  { nebula::meta::PatternMacro::DATE, "{date}" },
+  { nebula::meta::PatternMacro::HOUR, "{hour}" },
+  { nebula::meta::PatternMacro::MINUTE, "{minute}" },
+  { nebula::meta::PatternMacro::SECOND, "{second}" },
+  { nebula::meta::PatternMacro::TIMESTAMP, "{timestamp}" }
+};
+
```

Comment:
minor for discussion only, I don't have preference: instead of maintain multiple collection due to source format. I'm wondering if we can do single collection without format `/{` or `{`, just purely the MACRO NAME `DATE`, `HOUR`, `MINUTE`, and let the matcher to format the string when doing the search.

Reply:
agree, this seems overkill to avoid those macro , we can put a reserved words list.
like https://ci.apache.org/projects/flink/flink-docs-stable/dev/table/sql/#reserved-keywords

Reply:
55e454d use DATE HOUR MINUTE SECOND all cap as macro

Reply:
RESERVED KEYWORD is a good idea when applied in schema/coding, but restrict user arbitrary path (e.g s3) to not have normal words like DATE is too fragile I think. I think in file path, we should still enforce `{DATE}` format, but internally we can treat the MACRO name as reserved `DATE`. what do you think?